server/src/main/java/org/opensearch/index/store/lockmanager/FileLockInfo.java

```diff
@@ -97,13 +103,38 @@ public static String getFileToLockNameFromLock(String lockName) {
         }
 
         public static String getAcquirerIdFromLock(String lockName) {
-            String[] lockNameTokens = lockName.split(RemoteStoreLockManagerUtils.SEPARATOR);
+            if (lockName.endsWith(RemoteStoreLockManagerUtils.V1_LOCK_FILE_EXTENSION)) {
+                // this is the old lock file created for versions <= 2.10
+                return getAcquirerIdFromV1Lock(lockName);
+            }
+            String[] lockNameTokens = lockName.split(Pattern.quote(RemoteStoreLockManagerUtils.SEPARATOR));
 
             if (lockNameTokens.length != 2) {
                 throw new IllegalArgumentException("Provided Lock Name " + lockName + " is not Valid.");
             }
             return lockNameTokens[1].replace(RemoteStoreLockManagerUtils.LOCK_FILE_EXTENSION, "");
         }
+
+        private static String getFileToLockNameFromV1Lock(String lockName) {
+            if (lockName.lastIndexOf(RemoteStoreLockManagerUtils.V1_LOCK_SEPARATOR) == INVALID_INDEX
+                || lockName.lastIndexOf(RemoteStoreLockManagerUtils.V1_LOCK_FILE_EXTENSION) == INVALID_INDEX) {
+                throw new IllegalArgumentException("Provided Lock Name " + lockName + " is not Valid.");
+            }
+            String acquirerId = getAcquirerIdFromV1Lock(lockName);
+            return lockName.substring(0, lockName.lastIndexOf(acquirerId));
```

Comment:
this looks incorrect, as we will return the file name along with lock separator. Here we should just return `lockName.substring(0, lockName.lastIndexOf(RemoteStoreLockManagerUtils.V1_LOCK_SEPARATOR))`

Reply:
my bad.. updated